### PR TITLE
27 form builder nancy

### LINKED ISSUE #45: 27. Form Builder
Purpose: The goal is to add admin/super admin users to create forms
Function: To create questions.Have an add button, that will have a dropdown on the type of question they want, that will create a question 
You can add questions  and if it is multiple choice  you add options
Once done, there is an option to save the form
There is also  an option to push publish -> that will just make it public to client    users

### PR DESCRIPTION
**Pull request by nancytaen**
Edit on March 8th:
* includes merged changes with Youssef's and Sunny's branches
* TODO on error display on frontend and retrieving admin_id from redux, but I will address these next time since the pr is getting big

=====================================================

Major TODOs:
* support multiple choice/checkbox grids, linear scale, remove upload file option
* draggable button to allow questions/option order to be changed
* data format - numbering questions/options (?), visibility level
* data validation, frontend and backend
* react optimization - currently the entire form is passed down to children as props. use click outside / redux to reduce unnecessary rendering
* clean up code

### CODE REVIEW

File: Backend/app/controllers/auth.controller.js [hunk 1]
```diff
@@ -1,34 +1,74 @@
...
+      const { id } = verifyEmailJWT(verificationCode);
+      const verified = await Customer.verify(id);
+      res.json({ success: verified });
+    } catch (err) {
+      next(err);
+    }
+  },
+  async logout(req, res, next) {
+    try {
+      // TODO: Clear the localstorage of the token
```
Comment (ashih2018) [inline Backend/app/controllers/auth.controller.js:66]: Optional: Storing JWTs can be done in the frontend. You can also look at storing the token in cookies as an alternative

File: Backend/app/controllers/custom-form.controller.js [hunk 2]
```diff
@@ -1,44 +1,69 @@
...
+    } catch (err) {
+      next(err);
+    }
+  },
+
+  async updateCustomForm(req, res, next) {
+    try {
+      const { form_id } = req.params;
+      const form = new CustomForm({ form_id: form_id, ...req.body });
+      logger.debug('update controller');
```
Comment (ashih2018) [inline Backend/app/controllers/custom-form.controller.js]: Can you open a new issue to clean up the unused logger statements

**Review by ashih2018 — CHANGES_REQUESTED** [review]
Overall the backend changes look fine. In the future pls split up into multiple PRs (one for the JWT authentication, one for the querying, etc.) or it will make me sad :`)
Made a couple small comments but would also be super helpful to add comments on how to run/test, I couldn't end up running this on my machine. Also there are duplicated Frontend files that you should get rid of, let me know if you need help doing so!

File: Backend/app/controllers/auth.controller.js (hunk 1 above)
Comment (nancytaen) [reply Backend/app/controllers/auth.controller.js:66]: This part is actually done by @youssefsoli so he can address it better. I'm not sure why the changes are showing up on my pr, maybe I pulled from a branch ahead of main?

File: Backend/app/controllers/custom-form.controller.js (hunk 2 above)
Comment (nancytaen) [reply Backend/app/controllers/custom-form.controller.js]: Okay, will do

**Pull request merged**